### ISSUE TITLE
Complex Type Serialization via FluentAPI

### ISSUE THREAD

**Issue by Ma3yTa** [aspnet/EntityFramework6#72]
Is the way to define serialization/deserialization in OnModelCreating and leave only POCO class with one property ``public MyComplexType MyComplexType { get; set; }`` without attributes and additional property?

```csharp
public class MyEntity
{
    //...
    [NotMapped]
    public MyComplexType MyComplexType { get; set; }

    public string MySerializedComplexType
    {
        get { return Serialize(MyComplexType); }
        set { MyComplexType = Deserialize(value); }
    }
    //...
}
```
Thanks

**Comment by divega** [comment]
@Ma3yTa Just to confirm, are  you asking for the ability to register in the EF model the 2-way conversions from a string stored in a database column to a object instance? 

If that is what your question is about, the answer is that we are not planning to do it for EF6, and maybe we will do it in the future for EF Core. 

The following EF Core issues contain relevant discussions:
https://github.com/aspnet/EntityFramework/issues/240
https://github.com/aspnet/EntityFramework/issues/4021
https://github.com/aspnet/EntityFramework/issues/246

**Comment by Ma3yTa** [comment]
@divega yes. thanks

**Comment by Ma3yTa** [comment]
@divega Do you have plans to work on EF7 and implement those features http://enterprisecraftsmanship.com/2014/11/29/entity-framework-6-7-vs-nhibernate-4-ddd-perspective/ ?

**Comment by divega** [comment]
@Ma3yTa I only skimed over that article (which seems very good) and here is a summary:
- The mentioned "requirement" to use foreign key properties is not accurate: For EF Core you can have FKs in shadow state (meaning they are tracked by the DbContext but they don't appear in the objects). EF6 has alwas supported independent associations that don't have FKs, although they have limitations and we recommend having FK properties.
- EF Core already has some support for Hi/Lo an other value generators.
- AFAIR, the ability to map collection navigation properties to fields is not there. @ajcvickers may provide more inforation but I think this would be an easy gain to enable the kind of read-only collection pattern described in the article.

**Comment by ajcvickers** [comment]
@Ma3yTa @divega The current codebase (to be released as 1.1) supports using backing fields for navigation properties. However, we could probably improve the API surface. I'll file an issue on that.

Also, EF Core supports deleting orphans--point 1 in the post.

**Comment by Ma3yTa** [comment]
Very bad. We choose NHibernate for our project. But migrations is very good.
